api/src/lib/orders/src/get-orders.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * @param {ordersPgRepo} ordersRepo
+ */
+function GetordersFactory(deps) {
+    'use strict'
+
+    const { ordersRepo } = deps
+
+    const getOrders = async (ctx) => {
+        const logger = ctx.request.state.logger
+
+        var orderDetails=[];
+  
+
+        try {
+
+            const orders = await ordersRepo.get.byId(ctx.state.session.id)
+            for (var i = 0; i < orders.length;i++) {
+                var prods = orders[i].productids.split(',');
+                const products = await ordersRepo.get.prod(prods)
+                orderDetails.push({ id: orders[i].id, price: orders[i].totalprice, products: products, date: orders[i].purchasedate})
+
+            }
```

Comment:
Again, `var` should not be used here, because this loop includes asynchronous features, it _can't_ leverage functional programming the same way the other example did. In this case, to reduce the responsibilities this controller has, we can move the loop logic to another function:

```javascript
/**
 * @param {ordersPgRepo} ordersRepo
 */
function GetordersFactory(deps) {
  'use strict'

  const { ordersRepo } = deps

  /**
   * Gets the products associated with the given order, and returns a model 
   * that merges order and product data
   * @param {Order} order -- a single order by a customer
   * @returns {OrderHistory} -- the order with product data mapped to it
   */
  const mapOneOrderToProducts = async (order) => {
    const productIds = order.productids.split(',')
    const products = await ordersRepo.get.prod(productIds)
    return {  
      id: order.id,
      price: order.totalprice,
      products: products,
      date: order.purchasedate,
    }
  }

  /**
   * Gets the products associated with an array of orders, and returns the 
   * orders merged with product data
   * @param {Order[]} orders -- the orders by a customer
   * @param {OrderHistory[]?} orderDetails -- optional array to place the merged orders into
   * @returns {OrderHistory[]} -- the orders with product data mapped to them
   */  
  const mapOrdersToProducts = async (orders, orderDetails = []) => {
    for (let i = 0; i < orders.length; i += 1) {
      orderDetails.push(await mapOneOrderToProducts(orders[1]))
    }

    return orderDetails
  }

  /**
   * A route handler to retrieve a customer's orders
   * @param {KoaContext} ctx -- Koa's request context
   */
  const getOrders = async (ctx) => {
    const logger = ctx.request.state.logger

    try {
      const orders = await ordersRepo.get.byId(ctx.state.session.id)
      const orderDetails = await mapOrdersToProducts(orders)

      if (orderDetails) {
        ctx.response.status = 200
        ctx.response.body = orderDetails
      } else {
        ctx.response.status = 404
      }
    } catch (err) {
      logger.emit('product_read_error', 'error', { err })
      throw new Error('Failed to read product with uid ' + ctx.params.id)
    }
  }

  return { getOrders }
}

module.exports = GetordersFactory
```